### DISCUSSION TITLE
RTMP Pull and SRT Listener Publisher

### DISCUSSION THREAD

**Discussion by krisdahl** [OvenMediaLabs/OvenMediaEngine#1381] [Ideas]
Correct me if I’m wrong but currently there is not a publisher for RTMP pull or SRT in listener mode?

These are great options for bringing content into existing broadcast workflows, and wouldn’t require setting up pushes (which often require firewall rules and additional setup on receiving end)

**Comment by bchah** [comment]
+1 for this, related to #1408

**Comment by bchah** [comment]
Hi @getroot have you given this any consideration? An SRT Listener Publisher. It would be highly beneficial to support "plug and play" hardware decoder workflows 🙏

**Reply by getroot** [reply]
Although they are not high priority, these tasks are definitely registered in our backlog. (But there are so many higher priorities than these.)

**Comment by philipthomasme** [comment]
+1: An SRT Listener would turn this software into a very powerful tool for remote production workflows and therefore add a lot of value.

**Comment by bchah** [comment]
Users continue to ask for this every day so we had to find a workaround. I can confirm that [MediaMTX](https://github.com/bluenviron/mediamtx) works for this. We are deeply integrated with OME so the workflow is to push the SRT stream from OME to MediaMTX, then allow pull from MediaMTX.

Example of both running on one server (authentication not accounted for but there are plenty of options in MediaMTX):

SRT -> OME -> Start Streaming

OME Push API -> Push to srt://127.0.0.1:8890?streamid=publish:mystreamid

SRT Caller -> srt://publicserverIP:8890?streamid=read:mystreamid

It works great. Of course it would be great to see this happening natively in OME one day 🙏
